src/rgw/rgw_common.h

```diff
@@ -2423,4 +2441,37 @@ int decode_bl(bufferlist& bl, T& t)
   return 0;
 }
 
+static inline void start_trace(span_structure&& ss, Span&& span, req_state* const s, const char* name, bool should_store)
+{
+  #ifdef WITH_JAEGER
+    if(should_store)
+    {
+      Span span;
+      if(s && !s->stack_span.empty()){
+        span = tracer_2.child_span(name, s->stack_span.top());
```

Comment:
is tracer_2 thread safe?
since it is a global variable, you are going to access it from multiple threads

Reply:
yes tracer_2 is thread safe because it needs to be initialized only once and the rest of time it is serving as similar as a `read-only variable`

Reply:
yes it is thread safe and it is initialized only once and then acting as read-only variable for rest of the other request